matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java

```diff
@@ -0,0 +1,469 @@
+        }
+        for (MBranch mBranch : model.getBranches()) {
+            if (mBranch.getR() == 0 && mBranch.getX() == 0) {
+                vlGraph.addEdge(mBranch.getFrom(), mBranch.getTo());
+            }
+        }
+        for (Set<Integer> busNums : new ConnectivityInspector<>(vlGraph).connectedSets()) {
+            String voltageLevelId = "VL" + busNums.iterator().next();
+            containersMapping.voltageLevelIdToBusNums.put(voltageLevelId, busNums);
+            for (int busNum : busNums) {
+                containersMapping.busNumToVoltageLevelId.put(busNum, voltageLevelId);
+            }
+        }
+    }
+
+    private static String getBusId(int busNum) {
+        return "B" + busNum;
+    }
+
+    private static void createBuses(MatpowerModel model, ContainersMapping containerMapping, Network network, PerUnitContext perUnitContext) {
+        for (MBus mBus : model.getBuses()) {
+            String voltageLevelId = containerMapping.busNumToVoltageLevelId.get(mBus.getNumber());
+            String substationId = containerMapping.voltageLevelIdToSubstationId.get(voltageLevelId);
+
+            // create substation
+            Substation substation = createSubstation(network, substationId);
+
+            // create voltage level
+            VoltageLevel voltageLevel = createVoltageLevel(mBus, voltageLevelId, substation, network, perUnitContext);
+
+            // create bus
+            createBus(mBus, voltageLevel);
+
+            // create load
+            createLoad(mBus, voltageLevel);
+
+            // create shunt compensator
+            createShuntCompensator(mBus, voltageLevel, perUnitContext);
+
+            //create generators
+            createGenerators(model, mBus, voltageLevel);
+        }
+    }
+
+    private static void createGenerators(MatpowerModel model, MBus mBus, VoltageLevel voltageLevel) {
+        model.getGenerators().stream().filter(gen -> gen.getNumber() == mBus.getNumber()).forEach(mGen -> {
+            String busId = getBusId(mGen.getNumber());
+            String genId = createGenId(mGen, voltageLevel.getNetwork());
+            Generator generator = voltageLevel.newGenerator()
+                    .setId(genId)
+                    .setConnectableBus(busId)
+                    .setBus(isInService(mGen) ? busId : null)
+                    .setTargetV(mGen.getVoltageMagnitudeSetpoint() * voltageLevel.getNominalV())
+                    .setTargetP(mGen.getRealPowerOutput())
+                    .setTargetQ(mGen.getReactivePowerOutput())
+                    .setVoltageRegulatorOn(true)
+                    .setMaxP(mGen.getMaximumRealPowerOutput())
+                    .setMinP(mGen.getMinimumRealPowerOutput())
+                    .add();
+
+            generator.newMinMaxReactiveLimits()
+                    .setMinQ(mGen.getMinimumReactivePowerOutput())
+                    .setMaxQ(mGen.getMaximumReactivePowerOutput())
+                    .add();
```

Comment:
Should be move in the else block of the next `if` statement

Reply:
done